dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx

```diff
@@ -0,0 +1,212 @@
+
+  const newEntry = { label: "", aggregation: "", custom_label: "", name: "", side: "right" };
+
+  const [configList, setConfigList] = useState({
+    dimensions: [{ ...newEntry }],
+    metrics: [{ ...newEntry }],
+  });
+
+  useEffect(() => {
+    if (data.rawVizData?.dataConfig) {
+      setConfigList({
+        ...data.rawVizData?.dataConfig
+      })
+    } else if (data.defaultAxes.xaxis || data.defaultAxes.yaxis) {
+      const { xaxis, yaxis } = data.defaultAxes;
+      setConfigList({
+        dimensions: [...xaxis && xaxis],
+        metrics: [...yaxis && yaxis],
+      })
+    }
+  }, [data.defaultAxes, data.rawVizData?.dataConfig]);
+
+  const updateList = (value: string, index: number, name: string, field: string) => {
+    let list = { ...configList };
+    let listItem = { ...list[name][index] };
+    listItem = { ...listItem, [field]: value };
+    const newList = {
+      ...list,
+      [name]: [
+        ...list[name].slice(0, index),
+        listItem,
+        ...list[name].slice(index + 1, list[name].length),
+      ],
+    };
+    setConfigList(newList);
+  }
+
+  const onfieldOptionChange = (e, index: number, name: string) => {
+    let label = e.length > 0 ? e[0].label : '';
+    updateList(label, index, name, 'label');
+  };
+
+  const onAggregationChange = (e, index: number, name: string) => {
+    let label = e.length > 0 ? e[0].label : '';
+    updateList(label, index, name, 'aggregation');
+  };
+
+  const onCustomLabelChange = (e, index: number, name: string) => {
+    console.log('name ', name)
+    updateList(e.target.value, index, name, 'custom_label');
+  };
+
+  const handleSideChange = (id: string, index: number, name: string) => {
+    updateList(id, index, name, 'side');
+  }
+
+  const handleServiceRemove = (index: number, name: string) => {
+    const list = { ...configList };
+    const arr = [...list[name]];
+    arr.splice(index, 1);
+    const y = { ...list, [name]: arr }
+    setConfigList(y);
+  };
+
+  const handleServiceAdd = (name: string) => {
+    let newList = { ...configList, [name]: [...configList[name], newEntry] }
+    setConfigList(newList);
+  };
+
+  const updateChart = () => {
+    dispatch(
+      renderExplorerVis({
+        tabId: tabId,
+        data: {
+          ...explorerVisualizations,
+          dataConfig: {
+            metrics: configList.metrics,
+            dimensions: configList.dimensions
+          }
+        }
+      })
+    );
+  }
+
+  const getCommonUI = (lists, sectionName: string) => lists.map((singleField, index: number) => (
+    <>
+      <div key={index} className="services">
+        <div className="first-division">
+          <EuiPanel color="subdued">
+            <EuiFormRow
+              label="Aggregation"
+              labelAppend={
+                lists.length !== 1 && (
+                  <EuiText size="xs">
+                    <EuiIcon type="cross" color="danger" onClick={() => handleServiceRemove(index, sectionName)} />
```

Comment:
This should be multi line. You have 3 properties and you are also including a function inside one of them. Too much easier to read when done vertically.